go-controller/pkg/ovn/pods.go

```diff
@@ -524,18 +489,58 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 		return fmt.Errorf("failed to handle external GW check: %v", err)
 	}
 
+	// set addresses on the port
+	addresses = make([]string, len(podIfAddrs)+1)
+	addresses[0] = podMac.String()
+	for idx, podIfAddr := range podIfAddrs {
+		addresses[idx+1] = podIfAddr.IP.String()
+	}
+
+	// LSP addresses in OVN are a single space-separated value
+	cmd, err = oc.ovnNBClient.LSPSetAddress(portName, strings.Join(addresses, " "))
+	if err != nil {
+		return fmt.Errorf("unable to create LSPSetAddress command for port: %s", portName)
+	}
+	cmds = append(cmds, cmd)
+
+	// add external ids
+	extIds := map[string]string{"namespace": pod.Namespace, "pod": "true"}
+	cmd, err = oc.ovnNBClient.LSPSetExternalIds(portName, extIds)
+	if err != nil {
+		return fmt.Errorf("unable to create LSPSetExternalIds command for port: %s", portName)
+	}
+	cmds = append(cmds, cmd)
+
 	// CNI depends on the flows from port security, delay setting it until end
 	cmd, err = oc.ovnNBClient.LSPSetPortSecurity(portName, strings.Join(addresses, " "))
 	if err != nil {
 		return fmt.Errorf("unable to create LSPSetPortSecurity command for port: %s", portName)
 	}
 
-	err = oc.ovnNBClient.Execute(cmd)
+	cmds = append(cmds, cmd)
+
+	// execute all the commands together.
+	err = oc.ovnNBClient.Execute(cmds...)
 	if err != nil {
-		return fmt.Errorf("error while setting port security on port: %s error: %v",
+		return fmt.Errorf("error while creating logical port %s error: %v",
 			portName, err)
 	}
 
+	lsp, err = oc.ovnNBClient.LSPGet(portName)
+	if err != nil || lsp == nil {
+		return fmt.Errorf("failed to get the logical switch port: %s from the ovn client, error: %s", portName, err)
+	}
+
+	// Add the pod's logical switch port to the port cache
+	portInfo := oc.logicalPortCache.add(logicalSwitch, portName, lsp.UUID, podMac, podIfAddrs)
+
+	// If multicast is allowed and enabled for the namespace, add the port
+	// to the allow policy.
+	if oc.multicastSupport && multicastEnabled {
```

Comment:
@trozet since we're not holding the nsInfo lock while updating the MC stuff, an NS Update could come in and change multicastEnabled on the Namespace while we're still doing stuff in addLogicalPort(), before we get to podAddAloowMulticastPolicy().

Reply:
FWIW, I fixed this for Tim largely by ignoring the race for a couple reasons. First, instead of using the nsInfo.multicastEnabled we grab the value off the informer's NS object itself, which will be more up-to-date than the nsInfo object we got before doing a bunch of OVN transactions. Second, if the pod gets it wrong, the Namespace code will fix it later, and lastly mitigation is as easy as toggling MC on the NS and back. The scale issues are more pressing for now, and we have a plan to fix MC for real.